client_restaurant/shared/src/commonMain/kotlin/domain/usecase/ManageRestaurantInfoUseCase.kt

```diff
@@ -12,14 +13,16 @@ interface IManageRestaurantInfoUseCase {
     fun validateRestaurantDescription(description: String): Boolean
 }
 
-class ManageRestaurantInfoUseCase(private val remoteGateWay: IRemoteGateWay) :
+class ManageRestaurantInfoUseCase(private val remoteRestaurantGateway: IFakeRemoteGateWay) :
     IManageRestaurantInfoUseCase {
     override suspend fun getRestaurantInfo(restaurantId: String): Restaurant {
-        return remoteGateWay.getRestaurantInfo(restaurantId) ?: throw ServerSideException()
+        return remoteRestaurantGateway.getRestaurantInfo(restaurantId)
+            ?: throw ServerSideException()
     }
 
     override suspend fun updateRestaurantInfo(restaurant: Restaurant): Restaurant {
-        return remoteGateWay.updateRestaurantInfo(restaurant) ?: throw ServerSideException()
+        return remoteRestaurantGateway.updateRestaurantInfo(restaurant)
+            ?: throw ServerSideException()
```

Comment:
Throwing server exception in the domain layer!
The class/layer handling the connection with the server must throw this exception, not the domain

Reply:
I think this is just for now as it's a fake code till now, but thanks for notes.